op-service/sources/l1_beacon_client.go

```diff
@@ -129,6 +129,9 @@ func (cl *BeaconHTTPClient) BeaconBlobSideCars(ctx context.Context, fetchAllSide
 	if err := cl.apiReq(ctx, &resp, reqPath, reqQuery); err != nil {
 		return eth.APIGetBlobSidecarsResponse{}, err
 	}
+	if len(resp.Data) != len(hashes) {
+		return eth.APIGetBlobSidecarsResponse{}, fmt.Errorf("#returned blobs(%d) != #requested blobs(%d)", len(resp.Data), len(hashes))
+	}
```

Comment:
OK I get it now. We need to error earlier, of the fallback logic doesn't select the next beacon
```suggestion
	if len(resp.Data) < len(hashes) {
		return eth.APIGetBlobSidecarsResponse{}, fmt.Errorf("unexpected successful HTTP request with #returned blobs(%d) < #requested blobs(%d)", len(resp.Data), len(hashes))
	}
```
I believe an inequality is better here to keep `fetchAllSidecars` still working.

I believe we're still fixing a wrongly implemented Beacon API here, as a 404 should be returned not a 200 and an empty list.

Please also add a beacon fallback test for this case, where the first beacon API returns a 200+empty list and the fallback logic correctly moves on to the next beacon source.

Reply:
Good point, I pushed a fix to be compatible with `fetchAllSidecars`.

Reply:
Thanks, see my new review, I believe your old solution fixed it in a better spot. A new test case would still be great!

Reply:
It seems the `BeaconBlobSideCars` is mocked as a whole, there seems to be no way to test for this mocked function?